src/utils/hide-delete-status.ts

```diff
@@ -0,0 +1,5 @@
+import {UserInfo} from "../schemas/user.entity";
+
+export const hideDeleteStatus = (user: any) => {
+  return delete user.cart.isDeleted
```

Comment:
You seem to delete property from the original object, so you'd never have it again. It seems not the best way to handle it.